web/js/components/animation-widget/preload-spinner.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, ModalBody, ModalHeader } from 'reactstrap';
+import Spinner from 'react-loader';
+
+export default function PreloadSpinner({ onClose, title, bodyMsg }) {
+  return (
+    <Modal
+      isOpen
+      toggle={onClose}
+      size="sm"
+      backdrop={false}
+      wrapClassName="clickable-behind-modal"
+    >
+      <ModalHeader toggle={onClose}>{title}</ModalHeader>
+      <ModalBody>
+        {bodyMsg && (
+          <div style={{
+            minHeight: 30, marginBottom: 30, fontSize: 14,
```

Comment:
I kind of like the compactness of styles within smaller reusable files like this, but want to point out if not intended for any future reuse.

Reply:
I feel the same.  We don't normally do this but it's pretty common in some other React projects to have the related styles defined within the same component file.  I think this makes a lot of sense when the styles are for something like this, where it's not a set of styles that's going to be applied elsewhere, it's very specifically for parts of this component.

Reply:
@Benjaki2 I think that StackOverflow post is primarily focused on the performance argument of CSS vs inline styles and is about a decade old at this point.  Since then several popular libraries such as [Radium](https://github.com/FormidableLabs/radium) and [JSS](https://cssinjs.org/?v=v10.7.1) have been developed to try to get around this issue while also adding other powerful features.  I am not very familiar with the intricacies of either of these but at a glance they seem useful and, there seem to be some [solid arguments for ditching CSS classes in favor of styles defined in JS](https://speakerdeck.com/vjeux/react-css-in-js)

Regarding dynamic styles in JS, I think that's an even stronger argument for keeping the style declarations within the React component files themselves. It can be tedious trying to debug layout issues for components that are both being styled by a CSS stylesheet and inline, of which we have several examples.

That said, if you think it's worth moving these few style properties into a class definition within a CSS file for consistency I'm fine with doing so.

Reply:
Both approaches seem like viable options. Maybe we should make it a dev topic for next week. I don't mind keeping the code here as is. This seems like a bigger topic; should move to something like Radium or keep the old approach?

Reply:
Completely moving to inline styles at this point could be a big lift but, I think it's worth discussing whether or not we want to explore a new approach moving forward.